base/logging.go

```diff
@@ -374,10 +374,18 @@ func LogTraceEnabled(logKey LogKey) bool {
 func AssertLogContains(t *testing.T, s string, f func()) {
 	b := bytes.Buffer{}
 
-	// temporarily override logger output for the given function call
+	// Temporarily override logger output for the given function call
 	consoleLogger.logger.SetOutput(&b)
 	f()
+	// Allow time for logs to be printed
+	retry := func() (shouldRetry bool, err error, value interface{}) {
+		if strings.Contains(b.String(), s) {
+			return false, nil, nil
+		}
+		return true, nil, nil
+	}
+	err, _ := RetryLoop("wait for logs", retry, CreateSleeperFunc(10, 100))
```

Comment:
NITPICK feel free to ignore: if we're just waiting for that to happen, could this be `WaitForCondition`?

Reply:
Did think about that but didn't want it to rely on passing a RT. Will change the PR to use `WaitForConditionWithOptions` instead.